packages/ckb-indexer/src/collector.ts

```diff
@@ -0,0 +1,250 @@
+        validators.ValidateScript(type);
+        this.type = type;
+      } else if (typeof type === "object" && instanceOfScriptWrapper(type)) {
+        validators.ValidateScript(type.script);
+        this.type = type.script;
+      }
+    }
+
+    if (fromBlock) {
+      utils.assertHexadecimal("fromBlock", fromBlock);
+    }
+    if (toBlock) {
+      utils.assertHexadecimal("toBlock", toBlock);
+    }
+    if (order !== Order.asc && order !== Order.desc) {
+      throw new Error("Order must be either asc or desc!");
+    }
+    if (outputCapacityRange) {
+      utils.assertHexadecimal("outputCapacityRange[0]", outputCapacityRange[0]);
+      utils.assertHexadecimal("outputCapacityRange[1]", outputCapacityRange[1]);
+    }
+
+    if (outputDataLenRange) {
+      utils.assertHexadecimal("outputDataLenRange[0]", outputDataLenRange[0]);
+      utils.assertHexadecimal("outputDataLenRange[1]", outputDataLenRange[1]);
+    }
+
+    if (skip && typeof skip !== "number") {
+      throw new Error("Skip must be a number!");
+    }
+
+    if (sizeLimit && typeof sizeLimit !== "number") {
+      throw new Error("sizeLimit must be a number!");
+    }
+
+    this.indexer = indexer;
+    this.data = data;
+    this.fromBlock = fromBlock;
+    this.toBlock = toBlock;
+    this.order = order as Order;
+    this.skip = skip;
+    this.argsLen = argsLen as number;
+    this.outputCapacityRange = outputCapacityRange;
+    this.outputDataLenRange = outputDataLenRange;
+    this.sizeLimit = sizeLimit;
+  }
+
+  private hexStringPlus(hexString: Hexadecimal, addend: number): Hexadecimal {
+    const result = BigInt(hexString) + BigInt(addend);
+    return `0x${result.toString(16)}`;
+  }
+
+  //TODO change to input QueryOption type and return SearchKey Type
+  private generatorSearchKey(): SearchKey {
+    let script: Script | undefined = undefined;
+    const filter: SearchFilter = {};
+    let script_type: ScriptType | undefined = undefined;
+
+    if (this.lock) {
+      script = this.lock;
+      script_type = ScriptType.lock;
+      if (this.type && typeof this.type !== "string") {
+        filter.script = this.type;
+      }
+    } else if (this.type && typeof this.type !== "string") {
+      script = this.type;
+      script_type = ScriptType.type;
+    }
+    let block_range: HexadecimalRange | null = null;
+    if (this.fromBlock && this.toBlock) {
+      //this.toBlock+1 cause toBlock need to be included
+      block_range = [this.fromBlock, this.hexStringPlus(this.toBlock, 1)];
+    }
+    if (block_range) {
+      filter.block_range = block_range;
+    }
+    if (this.outputDataLenRange) {
+      filter.output_data_len_range = this.outputDataLenRange;
+    }
+    if (this.outputCapacityRange) {
+      filter.output_capacity_range = this.outputCapacityRange;
+    }
+    if (!script) {
+      throw new Error("Either lock or type script must be provided!");
+    }
+    if (!script_type) {
+      throw new Error("search type must be provided");
```

Comment:
```suggestion
      throw new Error("script_type must be provided");
```

we'd better keep the glossary consistent